### DISCUSSION TITLE
Storage Bookmarks

### DISCUSSION THREAD

**Discussion by scottkuhl** [CommunityToolkit/Maui#1548] [New Feature Discussions]
For my cross-platform .NET MAUI desktop application I want the user to pick a location to save their data, ideally with them picking a folder that is synchronized to the cloud for backup, like OneDrive.

The application will let them open files from any folder and retain a link to it to open it again in the future.

So over time I need to re-open data files stored in a given folder and re-open files that can be scattered across the file system.

For this I would use the .NET MAUI Community Toolkit FolderPicker, FileSaver and built in FilePicker.

It is my understanding that the folder and file permissions granted to the application do not persist across application sessions.

It seems the way to do this in a sandbox is by using Bookmarks.  See Avalonia's feature here: https://docs.avaloniaui.net/docs/next/concepts/services/storage-provider/bookmarks

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889 where it was agreed that New Features need to get above 7 upvotes within 6 months to progress to being reviewed by the maintainers.

**Comment by julianadormon** [comment]
@scottkuhl I would really love this as well, though I am late to the party, it seems. Did you happen to figure out a solution? Thanks.

**Reply by scottkuhl** [reply]
You have to handle it yourself with platform specific code. I'm really surprised the .NET MAUI team has not addressed this.  I would think this is a common problem.  But I guess most people just sync the data to their own cloud backend and let the app data live in the sandbox.  We ended up going with a Web Assembly solution instead.

This topic is covered on an episode of [Merge Conflict: 296: NSUrl, Security Scope, & Bookmarks](https://www.mergeconflict.fm/296).
